apis/core.oam.dev/v1alpha2/core_scope_types.go

```diff
@@ -47,8 +47,24 @@ type HealthScopeSpec struct {
 	// ProbeInterval is the amount of time in seconds between probing tries.
 	ProbeInterval *int32 `json:"probe-interval,omitempty"`
 
+	// AppRefs records references of applications' components
+	AppRefs AppReferences `json:"appReferences,omitempty"`
+
 	// WorkloadReferences to the workloads that are in this scope.
-	WorkloadReferences []corev1.ObjectReference `json:"workloadRefs"`
+	// +deprecated
+	WorkloadReferences []corev1.ObjectReference `json:"workloadRefs,omitempty"`
+}
+
+// AppReferences records references of applications' components
+type AppReferences map[string]CompReferences
+
+// CompReferences records references of components' resources
+type CompReferences map[string]ComponentReference
+
+// ComponentReference records references of resources of a component
+type ComponentReference struct {
+	Workload corev1.ObjectReference   `json:"workload,omitempty"`
+	Traits   []corev1.ObjectReference `json:"traits,omitempty"`
```

Comment:
Can a healthscope check the status of more than one trait?

Reply:
Yes, actually, it just how current app controller CUE-based health checking mechanism works.

Reply:
BTW, these traits references are not used now. CUE-based health checking has its own way to get traits references.
This field will be used for non-CUE based health checking in HealthScope controller in the future.